src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -382,13 +382,21 @@ static bool equalMetadataVectors(BSONObj& mdV1, BSONObj& mdV2)
 *
 * emptyAttributeValue -
 *
-* Check that the attribute doesn't have any value, taking into account its multi-valuated
-* nature
+* Check that the attribute doesn't have any value, taking into account its multi-valuated nature.
+*
+* NOTE
+*   1. ValueTypeNumber and ValueTypeBoolean are always non-empty.
+*   2. In API version 2, the value of a ContextAttribute CANNOT be left out
+*
 */
-bool emptyAttributeValue(ContextAttribute* caP)
+bool emptyAttributeValue(ContextAttribute* caP, const std::string& apiVersion)
 {
-  /* Note that ValueTypeNumber and ValueTypeBoolean are always non-empty */
-  return (caP->valueType == ValueTypeString) && (caP->stringValue == "") && (caP->compoundValueP == NULL);
+  if (apiVersion == "v1")
```

Comment:
I'm not sure but, as far as I understand, in v1 case the `ciP->apiVersion` (from where this `apiVersion` comes) could be `v1`, `ngsi10` or `ngsi9`. Thus, maybe is safer to revert the logical expression, i.e. `apiVersion != "v2"`).